code/datums/medal_awards.dm

```diff
@@ -1,66 +1,129 @@
+/// A list of all medals awarded to players; indexed by character real name.
 GLOBAL_LIST_EMPTY(medal_awards)
 
-
+/**
+ * A list of all medals awarded to a player.
+ */
 /datum/recipient_awards
-	var/list/medal_names
-	var/list/medal_citations
-	var/list/posthumous
+	/// Medal names.
+	var/list/medal_names = list()
+	/// Medal citations.
+	var/list/medal_citations = list()
+	/// If the medal was awarded after death.
+	var/list/posthumous = list()
+	/// The rank of the recipient.
 	var/recipient_rank
 
-/datum/recipient_awards/New()
-	medal_names = list()
-	medal_citations = list()
-	posthumous = list()
-
-
-/proc/give_medal_award(medal_location)
-	var/list/possible_recipients = list("Cancel")
-	var/list/listed_rcpt_ranks = list()
-	for(var/datum/data/record/t in GLOB.datacore.general)
-		var/rcpt_name = t.fields["name"]
-		listed_rcpt_ranks[rcpt_name] = t.fields["rank"]
-		possible_recipients += rcpt_name
-	var/chosen_recipient = tgui_input_list(usr, "Who do you want to award a medal to?", "Medal Recipient", possible_recipients)
-	if(!chosen_recipient || chosen_recipient == "Cancel") 
-		return
-	var/recipient_rank = listed_rcpt_ranks[chosen_recipient]
-	var/posthumous = 1
-	var/medal_type = tgui_input_list(usr, "What type of medal do you want to award?", "Medal Type", list("distinguished conduct medal", "bronze heart medal","medal of valor", "medal of exceptional heroism", "letter of commendation"))
-	if(!medal_type) 
-		return
-	var/citation = stripped_input(usr, "What should the medal citation read?", "Medal Citation")
-	if(!citation) 
-		return
-	for(var/mob/M in GLOB.alive_human_list)
-		if(M.real_name == chosen_recipient)
-			posthumous = 0
+/**
+ * Award a medal to a chosen marine.
+ * Does not check if the user has the right to award medals.
+ * - mob/user - The mob awarding the medal.
+ */
+/proc/do_award_medal(mob/user)
+	var/list/all_marines = list()
+	for(var/mob/living/mob as anything in GLOB.mob_living_list)
+		var/real_name = mob.real_name
+		var/datum/data/record/record
+		// holy shit theres gotta be a better way
+		for(var/datum/data/record/t in GLOB.datacore.general)
+			if(t.fields["name"] == real_name)
+				record = t
+				break
+		if(!record)
+			continue
+		all_marines[real_name] = list(GLOB.joined_player_list[real_name], record.fields["rank"])
```

Comment:
Human.get_paygrade()
if that fails then get_paygrades

See compose_job for example